store/src/java/com/zimbra/cs/filter/RuleManager.java

```diff
@@ -164,6 +175,81 @@ private static void setRules(Entry entry, String script, String sieveScriptAttrN
         }
     }
 
+    /**
+     * Reject the sieve script by throwing a ServiceException if its size exceeds
+     * the limit in the {@code zimbraMailSieveScriptMaxSize} attribute. If the new sieve
+     * script exceeds the limit, but is smaller than before, then the new script is not
+     * rejected.
+     *
+     * If the value of the attribute is not set, or if the attribute getter is not
+     * defined for the entry type, then the limit defined in the global config is used.
+     *
+     * @param entry the account/domain/cos/server for which the rules are to be saved
+     * @param newSieveScript the new sieve script, or an empty string
+     * @param sieveScriptAttrName the name of the sieve script attribute
+     * @throws ServiceException if the sieve script exceeds the value of the max size attribute
+     */
+    private static void rejectSieveScriptExceedingMaxSize(Entry entry, String newSieveScript, String sieveScriptAttrName)
+            throws ServiceException {
+        long sieveScriptMaxSize;
+        String zimbraMailSieveScriptMaxSize = entry.getAttr(Provisioning.A_zimbraMailSieveScriptMaxSize);
+        if (zimbraMailSieveScriptMaxSize != null && !zimbraMailSieveScriptMaxSize.isEmpty()) {
```

Comment:
If the value is `0` or an empty, the size is not limited. Can we run `return` to exit the function?
```
<attr id="9001" name="zimbraMailSieveScriptMaxSize" type="long" min="0" cardinality="single" optionalIn="globalConfig,account,cos,domain" flags="domainInfo,accountInfo,accountCosDomainInherited" since="11.0.0">
  <globalConfigValue>0</globalConfigValue>
  <defaultCOSValue>0</defaultCOSValue>
  <desc>Maximum size in bytes for sieve script attributes. When it is set to 0, the size is not limited.</desc>
</attr>
```

Reply:
done